packages/isar/lib/src/generator/code_gen/serialize_generator.dart

```diff
@@ -84,6 +84,11 @@ String _writeProperty({
           ? '$value$enumGetter?.toUtc().microsecondsSinceEpoch ?? $_nullLong'
           : '$value$enumGetter.toUtc().microsecondsSinceEpoch';
       return 'IsarCore.writeLong($writer, $index, $converted);';
+    case IsarType.duration:
+      final converted = nullable
+          ? '$value$enumGetter?.inMicroseconds ?? $_nullLong'
+          : '$value$enumGetter?.inMicroseconds';
+      return 'IsarCore.writeLong($writer, $index, $converted);';
```

Comment:
Just came across this PR and was wondering why we are using microseconds here if the docs say its milis. 

I am not deeply involved in isar, just curious :)

Reply:
You’d be correct, I am not deeply involved either. I got familiar enough with the code base to fix a pain point many had it seems. Ultimately we can either correct the documentation or change the code to use microseconds instead of milli.

This is purely a call to make and the trade off are precision vs max duration length on 64 bits. When designing this I made the call that most people would want to prefer longer duration and a millisecond precision than a shorter duration range a a microsecond precision.

Therefore I’d wager that the consistant thing to do is to correct the documentation and if possible commit message.

Thanks

Reply:
I've submitted to #1683 to correct this issue. The deserializer part was also aligned with the documentation, so it was clear that the original intent was indeed milliseconds and only the serializer had the typo.

Thanks for raising this @JulianBissekkou